### PR TITLE
Allow library users to configure Jetty directly

### PR DESCRIPTION
**Pull request by nasis**
@asafalima perhaps the cleanest way

### CODE REVIEW

**Comment by nasis** [comment]
@asafalima It's a very short PR...

File: src/main/java/io/logz/guice/jersey/JerseyModule.java [hunk 1]
```diff
@@ -5,22 +5,30 @@
...
 public class JerseyModule extends AbstractModule {
 
     private final JerseyConfiguration jerseyConfiguration;
+    private Consumer<Server> jettyConfigurer;
 
     public JerseyModule(JerseyConfiguration jerseyConfiguration) {
+        this(jerseyConfiguration, server -> {});
+    }
+
+    public JerseyModule(JerseyConfiguration jerseyConfiguration, Consumer<Server> jettyConfigurer) {
```
Comment (asafalima) [inline src/main/java/io/logz/guice/jersey/JerseyModule.java]: In order to let our users the possibility to configure the Jetty server and control the thread poll let's add the option to specify `JettyServerCreator` with default implementation of one that returns `new Server()` when using the old constructor.

Create interface named `JettyServerCreator` with a method `create`

File: src/main/java/io/logz/guice/jersey/JerseyModule.java [hunk 2]
```diff
@@ -5,22 +5,30 @@
...
         this.jerseyConfiguration = Objects.requireNonNull(jerseyConfiguration);
+        this.jettyConfigurer = jettyConfigurer;
```
Comment (asafalima) [inline src/main/java/io/logz/guice/jersey/JerseyModule.java]: Add `requireNonNull` and change the line above to use static import

File: src/main/java/io/logz/guice/jersey/JerseyServer.java [hunk 3]
```diff
@@ -28,9 +29,16 @@
 
     JerseyServer(JerseyConfiguration jerseyConfiguration,
                  Supplier<Injector> injectorSupplier) {
+        this(jerseyConfiguration, injectorSupplier, jettyServer -> {});
```
Comment (asafalima) [inline src/main/java/io/logz/guice/jersey/JerseyServer.java]: Remove old constructor, this is internal class that should only be initialized using the guice

File: src/main/java/io/logz/guice/jersey/JerseyServer.java [hunk 4]
```diff
@@ -28,9 +29,16 @@
...
+    }
+
+    JerseyServer(JerseyConfiguration jerseyConfiguration,
+                 Supplier<Injector> injectorSupplier,
+                 Consumer<Server> jettyConfigurer) {
         this.jerseyConfiguration = jerseyConfiguration;
         this.injectorSupplier = injectorSupplier;
         this.server = new Server();
+        jettyConfigurer.accept(server);
```
Comment (asafalima) [inline src/main/java/io/logz/guice/jersey/JerseyServer.java]: Here use the `JettyServerCreator::create`

**Pull request merged**